SPA QS backlinks to Dev Center (#10194)
auth0/docs: articles/quickstart/spa/react/01-login.md

---
<!-- markdownlint-disable MD002 MD034 MD041 -->

:::note

> For a deep dive into implementing user authentication in React, visit the [React Authentication By Example](https://developer.auth0.com/resources/guides/spa/react/basic-authentication/). This guide provides you with additional details, such as creating a signup button, protecting routes using different strategies, and using class components. 

Reviewer: ```suggestion
Visit the [React Authentication By Example](https://developer.auth0.com/resources/guides/spa/react/basic-authentication/) guide for a deep dive into implementing user authentication in React. This guide provides additional details on how to create a sign-up button and add route guards using React Router.
```

The `user` property contains sensitive information and artifacts related to the user's identity. As such, its availability depends on the user's authentication status. To prevent any render errors, use the `isAuthenticated` property from `useAuth0()` to check if Auth0 has authenticated the user before React renders any component that consumes the `user` property. Ensure that the SDK has completed loading before accessing the `isAuthenticated` property, by checking that `isLoading` is `false`.

:::panel Checkpoint

> Verify that you can display the `user.name` or [any other `user` property](https://auth0.com/docs/users/references/user-profile-structure#user-profile-attributes/) within a component correctly after you have logged in.

Reviewer: ```suggestion
Verify that you can display the `user.name` or [any other `user` property](https://auth0.com/docs/users/references/user-profile-structure#user-profile-attributes) within a component correctly after you have logged in.
```
